### DISCUSSION TITLE
Copilot's judgement of free access is unfair to open-source contributors

### DISCUSSION THREAD

**Discussion by reorx** [community/community#30734] [Copilot Conversations]
I'm a self-employed programmer who actively participates in the open-source community, my projects include Python packages ([httpstat](https://github.com/reorx/httpstat) has 5k stars) and several plugins for Obsidian, so I think I may be qualified for the requirements of using Copilot for free. Plus that I mainly use Copilot to write open source projects in VSCode, what I do for living is writing Python in Vim.

My situation: The price is currently a little bit higher for me since I'm still struggling to get a stable income. I believe it'll be OK for next year when my ITIN is approved, so I can open GitHub Sponsors and bring donations to balance the expense.

I've sent a ticket to GitHub in June, the support said:

> At the moment, 12 months of free use of Copilot is only available for people who maintain popular open-source projects

This is where I got confused, if Copilot requires everyone to pay to use, I would definitely not write this.
I know some people that have been given free use of Copilot for one year, they are confused because the stars of their repository are far less from being popular. I have also found some tweets (like https://twitter.com/niceoe/status/1539882356768509952 and https://twitter.com/ttmokeyjay/status/1539503239368159232),
the authors claim that they only have projects of about 200+ stars, yet they are eligible for the free use of Copilot.

Another thing support told me is that:

> We haven't invited absolutely everyone who qualifies for free Copilot usage all at once, because we are trying to make sure that we have no capacity issues.

Now it's about 2 months and all the enrollment should be over, I still don't get the 1 year free usage for Copilot.

To make it clear, here are my questions in brief:

1. some people I know with stars less than mine, or some people I found on twitter with less than 300 stars, are eligible to use Copilot for 12 months free. Why are people who have more contributions to the open-source community not qualified?
2. in the last ticket the support told me to wait for something, so I continued to wait for 2 months, yet I still don't have the access. Can the support check my account to see what's going on and how long should I wait further?

All I ask for is equal rights for what I've contributed to the open-source community in the past decade, which is also part of the contribution that Copilot could be made.

Thank you for reading this.

**Comment by Gomah** [comment]
100% agree with you @reorx.

I don't want to be off topic, but Github should review who should have access to **Copilot** for free.

I contribute to open-source by sponsoring independent maintainers, I've spent thousands of dollars to date in sponsorships and I thought Github would at least reward sponsors with a free Copilot membership just like open-source maintainers.

I know it's only $10 but the whole narrative around "free for open-source contributors" is quite off-putting.

**Comment by EnixCoda** [comment]
Thanks @reorx 

I also have projects with thousands of stars and >10k users. Being charged for Copilot is like being disqualified by GitHub as a maintainer of popular open-source projects. LOL

Although $10/m is not too expensive, Copilot saved me a lot of time and it really worth the price. But paying for it myself makes me feel bad as long as it is stated to be free for _people who maintain popular open-source projects_ and no clear definition of popular has been declared.

**Comment by foamzou** [comment]
Agree with @reorx 

I raise a support ticket for the request as well. And the support told me that

> We don't have a firm definition of "popular open source project" to share because we're expanding it as we go.
> Because we are trying to make sure that we have no capacity issues and Copilot works smoothly for everyone that's using it, our initial eligibility criteria has been applied to the first group of users, but we plan to expand the criteria and bring more people on board in the future.
> We are opening it up to more users as quickly as we can.

I don't know how long would take to the `future`.

**Comment by long2ice** [comment]
So what is real open-source contributors?

**Comment by monkeyWie** [comment]
Same here, I have two projects that star > 2k, but i'm not eligible to use Copilot.
Of course I have now spent 100$ on a year subscription and i'm just very curious  what is real open-source contributors?
